### PR TITLE
Add cycle7

### PR DESCRIPTION
**Pull request by EasonQian1**
## Description
Add cycle7

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/eason/cycle7/info.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            "command": false,
+            "nkro": false,
+            "backlight": false,
+            "rgblight": true,
+            "audio": false,
+            "key_lock": false
+        },
+        "rgblight": {
+            "pin": "B15",
+            "led_count": 0,
```
Comment (zvecr) [inline keyboards/eason/cycle7/info.json:48]: zero would be invalid, and previous config states `8`
```suggestion
            "led_count": 8,
```

File: keyboards/eason/cycle7/info.json [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            "hue_steps": 8,
+            "saturation_steps": 8,
+            "brightness_steps": 8,
+            "max_brightness": 255,
```
Comment (zvecr) [inline keyboards/eason/cycle7/info.json:52]:
```suggestion
```

File: keyboards/eason/cycle7/info.json [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            "bootmagic": true,
+            "mousekey": true,
+            "extrakey": true,
+            "console": false,
+            "command": false,
+            "nkro": false,
+            "backlight": false,
+            "rgblight": true,
+            "audio": false,
+            "key_lock": false
```
Comment (zvecr) [inline keyboards/eason/cycle7/info.json:44]:
```suggestion
            "bootmagic": true,
            "mousekey": true,
            "extrakey": true,
            "console": false,
            "command": false,
            "nkro": false,
            "rgblight": true
```

File: keyboards/eason/cycle7/halconf.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/bt66tech/bt66tech60/halconf.h -r platforms/chibios/common/configs/halconf.h`
+ */
+ #pragma once
+
+ #define HAL_USE_PWM TRUE
```
Comment (zvecr) [inline keyboards/eason/cycle7/halconf.h:23]:
```suggestion
```

File: keyboards/eason/cycle7/config.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#define RGB_DI_PIN B15
+#define RGBLED_NUM 8
```
Comment (zvecr) [inline keyboards/eason/cycle7/config.h:32]:
```suggestion
```

File: keyboards/eason/cycle7/config.h [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+#define RGBLIGHT_EFFECT_BREATHING
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
```
Comment (zvecr) [inline keyboards/eason/cycle7/config.h:30]: Should be in info.json
```suggestion

```

File: keyboards/eason/cycle7/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    make eason/cycle7:default
+
+Flashing example for this keyboard:
+
+    make eason/cycle7:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader Enter the bootloader in 3 ways:
+* **Bootmagic reset**: Hold down Enter in the keyboard then replug
```
Comment (zvecr) [inline keyboards/eason/cycle7/readme.md:20]: Given the default config of `0,0`, Enter would not be the bootmagic key location

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/eason/cycle7/readme.md (hunk 7 above)
Comment (EasonQian1) [reply keyboards/eason/cycle7/readme.md]: I have edited it, thanks a lot for your help！

File: keyboards/eason/cycle7/info.json [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+            "command": false,
+            "nkro": false,
+            "rgblight": true
+        },
+        "rgblight": {
+            "pin": "B15",
+            "led_count": 8,
+            "sleep": true,
+            "animations": {
+                "all": true
```
Comment (zvecr) [inline keyboards/eason/cycle7/info.json:48]: `all` is now considered invalid and effects must be explicitly listed/enabled

File: keyboards/eason/cycle7/config.h [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#define RGB_DI_PIN B15
```
Comment (zvecr) [inline keyboards/eason/cycle7/config.h:18]:
```suggestion

```

File: keyboards/eason/cycle7/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,_______, _______),
+  [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,_______,_______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______, _______,_______),
```
Comment (zvecr) [inline keyboards/eason/cycle7/keymaps/via/keymap.c:44]:
```suggestion
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______, _______,_______)
```

File: keyboards/eason/cycle7/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+   KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,  KC_SLSH, KC_RGHT,  KC_UP,
+   KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_SPC,  KC_SPC,  KC_SPC,  KC_SPC,  KC_SPC,    KC_SPC,  KC_RALT, KC_RGUI, MO(1), KC_RCTL, KC_LEFT, KC_DOWN
+ ),
+  [1] = LAYOUT(
+    KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______,_______,QK_BOOT,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,  _______,_______,_______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______, _______,_______),
```
Comment (zvecr) [inline keyboards/eason/cycle7/keymaps/default/keymap.c:32]:
```suggestion
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,_______,_______, _______,_______)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/eason/cycle7/info.json [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+            "bootmagic": true,
+            "mousekey": true,
+            "extrakey": true,
+            "console": false,
+            "command": false,
+            "nkro": false,
+            "rgblight": true
+        },
+        "rgblight": {
+            "pin": "B15",
```
Comment (drashna) [inline keyboards/eason/cycle7/info.json:44]: This is being migrated to `ws2812.pin` on develop.

File: keyboards/eason/cycle7/info.json [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+            "breathing": true,
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        }
```
Comment (drashna) [inline keyboards/eason/cycle7/info.json:58]:
```suggestion
                "alternating": true,
                "breathing": true,
                "christmas": true,
                "knight": true,
                "rainbow_mood": true,
                "rainbow_swirl": true,
                "rgb_test": true,
                "snake": true,
                "static_gradient": true,
                "twinkle": true
            }
```

File: keyboards/eason/cycle7/info.json (hunk 12 above)
Comment (EasonQian1) [reply keyboards/eason/cycle7/info.json]: I have already edited it.Thanks！:)

File: keyboards/eason/cycle7/info.json [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+                "A1",
+                "A0",
+                "C15",
+                "C14",
+                "B9",
+                "B8", "B7", "B6",
+                "A7",
+                "A6",
+                "A5",
+                "A4", "B5", "B4"]
```
Comment (drashna) [inline keyboards/eason/cycle7/info.json:31]:
```suggestion
            "rows": [ "A8",  "A9", "A10", "B3", "A15"],
            "cols": [ "A3", "A2", "A1", "A0", "C15", "C14", "B9", "B8", "B7", "B6", "A7", "A6", "A5", "A4", "B5", "B4"]
```

File: keyboards/eason/cycle7/info.json [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+            "bootmagic": true,
+            "mousekey": true,
+            "extrakey": true,
+            "console": false,
+            "command": false,
+            "nkro": false,
+            "rgblight": true
+        },
+        "ws2812": {
+        "pin": "B15"
```
Comment (drashna) [inline keyboards/eason/cycle7/info.json:44]:
```suggestion
            "pin": "B15"
```

File: keyboards/eason/cycle7/info.json [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+            "mousekey": true,
+            "extrakey": true,
+            "console": false,
+            "command": false,
+            "nkro": false,
+            "rgblight": true
+        },
+        "ws2812": {
+            "pin": "B15"
+            "led_count": 8,
```
Comment (drashna) [inline keyboards/eason/cycle7/info.json:29]: I believe you mean this? 
```suggestion
        "ws2812": {
            "pin": "B15"
        },
        "rgblight": {
            "led_count": 8,
```

File: keyboards/eason/cycle7/config.h [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#define WS2812_SPI SPID2
+#define WS2812_SPI_MOSI_PAL_MODE 5
+
+#define FORCE_NKRO
```
Comment (waffle87) [inline keyboards/eason/cycle7/config.h:21]: If NKRO isn't enabled to begin with, this should be removed.
```suggestion
```

File: keyboards/eason/cycle7/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+        KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_SLSH, KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_LALT, KC_LGUI, KC_MENU, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    )
+    [1] = LAYOUT(
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_1,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT,
+        KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_SLSH, KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_LALT, KC_LGUI, KC_MENU, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    )
```
Comment (waffle87) [inline keyboards/eason/cycle7/keymaps/default/keymap.c:33]: This won't work.
```suggestion
    )
```

File: keyboards/eason/cycle7/keymaps/via/keymap.c [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+        KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_SLSH, KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_LALT, KC_LGUI, KC_MENU, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    )
+    [1] = LAYOUT(
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_1,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT,
+        KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT, KC_SLSH, KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_LALT, KC_LGUI, KC_MENU, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    )
```
Comment (waffle87) [inline keyboards/eason/cycle7/keymaps/via/keymap.c:33]:
```suggestion
    )
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by EasonQian1** [comment]
Hi, bro. Can this PR be passed?@waffle87

File: keyboards/eason/cycle7/info.json [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+                {"label": "K4B (A15,A6)", "matrix": [4, 11], "x": 11.25, "y": 4, "w": 1.25},
+                {"label": "K4C (A15,A5)", "matrix": [4, 12], "x": 12.5, "y": 4, "w": 1.25},
+                {"label": "K4D (A15,A4)", "matrix": [4, 13], "x": 13.75, "y": 4, "w": 1.25},
+                {"label": "K4E (A15,B5)", "matrix": [4, 14], "x": 15.25, "y": 4},
+                {"label": "K4F (A15,B4)", "matrix": [4, 15], "x": 16.25, "y": 4},
+                {"label": "K3E (B3,B5)", "matrix": [3, 14], "x": 17.25, "y": 4}
+           ]
+            }
+        },
+    }
```
Comment (dunk2k) [inline keyboards/eason/cycle7/info.json:127]: formatting applied
```suggestion
{
    "keyboard_name": "Cycle7",
    "manufacturer": "Eason",
    "url": "",
    "maintainer": "EasonQian1",
    "usb": {
        "vid": "0x9A94",
        "pid": "0x9F70",
        "device_version": "1.0.0"
    },
    "processor": "STM32F103",
    "bootloader": "stm32duino",
    "matrix_pins": {
        "rows": [ "A8",  "A9", "A10", "B3", "A15"],
        "cols": [ "A3", "A2", "A1", "A0", "C15", "C14", "B9", "B8", "B7", "B6", "A7", "A6", "A5", "A4", "B5", "B4"]
    },
    "diode_direction": "ROW2COL",
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true,
        "console": false,
        "command": false,
        "nkro": false,
        "rgblight": true
    },
    "ws2812": {
        "pin": "B15"
    },
    "rgblight": {
        "led_count": 8,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
    },
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"label": "K00 (A8,A3)", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "K01 (A8,A2)", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "K02 (A8,A1)", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "K03 (A8,A0)", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "K04 (A8,C15)", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "K05 (A8,C14)", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "K06 (A8,B9)", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "K07 (A8,B8)", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "K08 (A8,B7)", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "K09 (A8,B6)", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": "K0A (A8,A7)", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "K0B (A8,A6)", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "K0C (A8,A5)", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "K0D (A8,A4)", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "K47 (A15,B8)", "matrix": [4, 7], "x": 14, "y": 0},
                {"label": "K0E (A8,B5)", "matrix": [0, 14], "x": 15.25, "y": 0},
                {"label": "K0F (A8,B4)", "matrix": [0, 15], "x": 16.25, "y": 0},
                {"label": "K2E (A10,B5)", "matrix": [2, 14], "x": 17.25, "y": 0},
                {"label": "K10 (A9,A3)", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "K11 (A9,A2)", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "K12 (A9,A1)", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "K13 (A9,A0)", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "K14 (A9,C15)", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "K15 (A9,C14)", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "K16 (A9,B9)", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "K17 (A9,B8)", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "K18 (A9,B7)", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "K19 (A9,B6)", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "K1A (A9,A7)", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "K1B (A9,A6)", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "K1C (A9,A5)", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "K1D (A9,A4)", "matrix": [1, 13], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "K1E (A9,B5)", "matrix": [1, 14], "x": 15.25, "y": 1},
                {"label": "K1F (A9,B4)", "matrix": [1, 15], "x": 16.25, "y": 1},
                {"label": "K2F (A10,B4)", "matrix": [2, 15], "x": 17.25, "y": 1},
                {"label": "K20 (A10,A3)", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "K21 (A10,A2)", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "K22 (A10,A1)", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "K23 (A10,A0)", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "K24 (A10,C15)", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "K25 (A10,C14)", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "K26 (A10,B9)", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "K27 (A10,B8)", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K28 (A10,B7)", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "K29 (A10,B6)", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": "K2A (A10,A7)", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "K2B (A10,A6)", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "K2C (A10,A5)", "matrix": [2, 12], "x": 12.75, "y": 2},
                {"label": "K2D (A10,A4)", "matrix": [2, 13], "x": 13.75, "y": 2, "w": 1.25},
                {"label": "K30 (B3,A3)", "matrix": [3, 0], "x": 0, "y": 3, "w": 1.25},
                {"label": "K31 (B3,A2)", "matrix": [3, 1], "x": 1.25, "y": 3},
                {"label": "K32 (B3,A1)", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "K33 (B3,A0)", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "K34 (B3,C15)", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "K35 (B3,C14)", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "K36 (B3,B9)", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "K37 (B3,B8)", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "K38 (B3,B7)", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "K39 (B3,B6)", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": "K3A (B3,A7)", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "K3B (B3,A6)", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "K3C (B3,A5)", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "K3D (B3,A4)", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "K3F (B3,B4)", "matrix": [3, 15], "x": 16.25, "y": 3},
                {"label": "K40 (A15,A3)", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.25},
                {"label": "K41 (A15,A2)", "matrix": [4, 1], "x": 1.25, "y": 4, "w": 1.25},
                {"label": "K42 (A15,A1)", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.25},
                {"label": "K46 (A15,B9)", "matrix": [4, 6], "x": 3.75, "y": 4, "w": 6.25},
                {"label": "K4A (A15,A7)", "matrix": [4, 10], "x": 10, "y": 4, "w": 1.25},
                {"label": "K4B (A15,A6)", "matrix": [4, 11], "x": 11.25, "y": 4, "w": 1.25},
                {"label": "K4C (A15,A5)", "matrix": [4, 12], "x": 12.5, "y": 4, "w": 1.25},
                {"label": "K4D (A15,A4)", "matrix": [4, 13], "x": 13.75, "y": 4, "w": 1.25},
                {"label": "K4E (A15,B5)", "matrix": [4, 14], "x": 15.25, "y": 4},
                {"label": "K4F (A15,B4)", "matrix": [4, 15], "x": 16.25, "y": 4},
                {"label": "K3E (B3,B5)", "matrix": [3, 14], "x": 17.25, "y": 4}
            ]
        }
    }
}
```

File: keyboards/eason/cycle7/info.json
Comment (dunk2k) [inline keyboards/eason/cycle7/info.json:1]: Looking at `LAYOUT` it appears there could be more than one physical key arrangement for this board. Could you append/amended `layouts` if this suspicion is correct?

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/eason/cycle7/info.json
Comment (dunk2k) [inline keyboards/eason/cycle7/info.json:1]: Hi @EasonQian1 

Could you confirm/correct matrix co-ordinates for the following key positions?:
- 2u Backspace: [0,13]
- 2.75u Right Shift: [3,12]
- ANSI Enter: [2,13]
- ISO Enter: [2,13]

File: keyboards/eason/cycle7/config.h [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#define WS2812_SPI SPID2
+#define WS2812_SPI_MOSI_PAL_MODE 5
```
Comment (fauxpark) [inline keyboards/eason/cycle7/config.h:19]:
```suggestion
#define WS2812_SPI_DRIVER SPID2
```

File: keyboards/eason/cycle7/info.json [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": false,
+        "rgblight": true
+    },
+    "ws2812": {
+        "pin": "B15"
```
Comment (fauxpark) [inline keyboards/eason/cycle7/info.json:28]:
```suggestion
        "pin": "B15",
        "driver": "spi"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**